crates/core/src/db/datastore/locking_tx_datastore/committed_state.rs

```diff
@@ -267,10 +268,12 @@ impl CommittedState {
             .get_mut(&table_id)
             .ok_or_else(|| TableError::IdNotFoundState(table_id))?;
         let blob_store = &mut self.blob_store;
-        table
-            .delete_equal_row(blob_store, rel)
-            .map_err(TableError::Insert)?
-            .ok_or_else(|| anyhow!("Delete for non-existent row when replaying transaction"))?;
+
+        // NOTE: We are eating the error here because during replay, it is assumed that
+        // the order of operations within a transaction mattered prior to mem-arch
+        // and that deleting and inserting the same row cannot occur in the commitlog
+        // after mem-arch.
```

Comment:
```suggestion
        // the order of operations within a transaction does not matter
        // and that deleting and inserting the same row cannot occur in the commitlog.
```

It's not clear to me that or how this changed, namely the [TxState invariants](https://github.com/clockworklabs/SpacetimeDB/blob/master/crates/core/src/db/datastore/locking_tx_datastore/tx_state.rs#L36-L40). 

In any case, for a future reader it likely only matters that the order does not matter.